tests/components/aladdin_connect/test_init.py

```diff
@@ -133,3 +135,38 @@ async def test_load_and_unload(
     assert await config_entry.async_unload(hass)
     await hass.async_block_till_done()
     assert config_entry.state == ConfigEntryState.NOT_LOADED
+
+
+async def test_stale_device_removal(
```

Comment:
Ideally I think this should set up the integration, add a device which doesn't exist (assert there is now two devices). Reload the integration and see that one disappears.

Reply:
Makes sense - I think this is what you had in mind...